csharp/src/Ice/OutgoingAsync.cs

```diff
@@ -441,6 +534,13 @@ public void RetryException()
                     InvokeExceptionAsync();
                 }
             }
+            catch (Ice.RemoteException ex)
+            {
+                if (Exception(ex))
+                {
+                    InvokeExceptionAsync();
+                }
+            }
```

Comment:
I don't see why `RemoteException` would be raised here.

Reply:
Fixing.